keda/templates/manager/clusterrole.yaml

```diff
@@ -26,77 +30,59 @@ rules:
   - events
   verbs:
   - '*'
+{{- if not .Values.permissions.operator.restrict.secret }}
 - apiGroups:
   - ""
-  resources:
-  - external
-  - pods
-    {{- if eq .Values.permissions.operator.restrict.secret false }}
+  resources:    
   - secrets
-    {{- end }}
-  - services
-  verbs:
-  - get
-  - list
-  - watch
-- apiGroups:
-  - ""
-  resources:
-  - limitranges
   verbs:
   - list
   - watch
+  {{- with .Values.permissions.operator.restrict.namesAllowList }}
 - apiGroups:
   - ""
   resources:
-  - serviceaccounts
-  verbs:
-  - list
-  - watch
-- apiGroups:
-  - '*'
-  resources:
-  - '*'
+  - secrets
   verbs:
   - get
+  resourceNames: {{ toYaml . | nindent 4 }}
+  {{- end }}
+{{- end }}
+
+{{- if .Values.rbac.enabledCustomScaledRefKinds }}
+{{- if not .Values.rbac.scaledRefKinds }}
 - apiGroups:
   - '*'
   resources:
   - '*/scale'
   verbs:
-  - get
-  - list
-  - patch
-  - update
-  - watch
-  {{- if and .Values.certificates.autoGenerated ( not .Values.certificates.certManager.enabled ) }}
+  - '*'
+{{- else }}
 - apiGroups:
-  - admissionregistration.k8s.io
+  - apps
   resources:
-  - validatingwebhookconfigurations
+  - deployments/scale
+  - statefulsets/scale
   verbs:
-  - get
-  - list
-  - patch
-  - update
-  - watch
+  - '*'
```

Comment:
Same as above, with this change we are getting the `delete` permission when it's not required